### PR TITLE
house number validation

### PR DESCRIPTION
**Pull request by katrinamariehh**
Added validation for checking that house numbers are positive integers for this [Pivotal story](https://www.pivotaltracker.com/story/show/126604869). Validation function is in value_format.clj; it uses the all-digits check first, then coerces the string to an integer and confirms that the integer is greater than or equal to zero. It's tested in value_format_test.clj (I included a test for all-digits that I wrote to understand testing and since there wasn't already one I left it).

### CODE REVIEW

File: src/vip/data_processor/validation/data_spec/value_format.clj [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
 (def all-digits
   {:check #"\A\d+\z"
    :message "Invalid data type"
    :severity :fatal})
 
+(defn check-not-negative [val]
+  (let [house-number (coerce/coerce-integer val)]
+      (and
+       (re-find (:check all-digits) val)
+       (>= house-number 0))))
```
Comment (tie-rack) [inline src/vip/data_processor/validation/data_spec/value_format.clj]: If `coerce-integer` returns `nil`, which it can, then this line with throw an exception because you can't compare `nil` to 0.

Comment (katrinamariehh) [inline src/vip/data_processor/validation/data_spec/value_format.clj]: To answer you first question--that's totally what I thought at first, too!  And then @sduckett was all like, "But what about NEGATIVE NUMBERS."  So this is specifically to make sure all the house numbers are positive integers.  Your second question about nil, if `(re-find (:check all-digits) val)` is `false` it doesn't evaluate the second part of the `and` and just returns `false`.

Comment (tie-rack) [inline src/vip/data_processor/validation/data_spec/value_format.clj]: You're right, it'll short-circuit out first... but there's still no way to have the `all-digits` check succeed and end up with a negative number for the `>=` to fail on. It's redundant.

Comment (katrinamariehh) [inline src/vip/data_processor/validation/data_spec/value_format.clj]: Now that I look at the test I think you're right; I thought I put test data through and it did not fail, but now I'm just not sure.  I'll take a look and get back to you!

Comment (katrinamariehh) [inline src/vip/data_processor/validation/data_spec/value_format.clj]: You're totally right! I *thought* I had checked before starting about negative integers, but clearly I did not!

Comment (katrinamariehh) [inline src/vip/data_processor/validation/data_spec/value_format.clj]: I'm going to remove my changes but add tests to confirm the expected behavior (that negative house numbers will be caught).
